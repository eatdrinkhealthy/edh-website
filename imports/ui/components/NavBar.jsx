import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

class HashLink extends Component {
  static propTypes = {
    to: PropTypes.string.isRequired,
  };

  handleClick = () => {
    document.querySelector(this.props.to).scrollIntoView();
  };

  render() {
    return <NavLink {...this.props} onClick={this.handleClick} />;
  }
}

const NavBar = () => (
  <div className="row">
    <div className="column">
      <div className="app-header__top">
        <a href="/">
          <img src="images/logo.png" alt="Company Logo" />
        </a>

        <nav className="app-header-nav">
          <ul className="menu align-center">
            <li>
              <NavLink to="/" activeStyle={{ color: "#f28020" }} exact>
                HOME
              </NavLink>
            </li>
            <li>
              <HashLink to="#contactForm" activeStyle={{ color: "#f28020" }} exact>
                CONTACT
              </HashLink>
            </li>
            <li>
              <NavLink to="/privacy-policy" activeStyle={{ color: "#f28020" }} exact>
                PRIVACY
              </NavLink>
            </li>
          </ul>
        </nav>

        <div className="app-header-social text-center medium-text-right">
          <a href="mailto:info@eatdrinkhealthy.co">
            <span className="fa-stack fa-lg">
              <i className="fa fa-circle fa-stack-2x" />
              <i className="fa fa-envelope fa-stack-1x fa-inverse" />
            </span>
          </a>
          <a
            href="https://www.linkedin.com/company/16173480/"
            target="_blank"
            rel="noopener noreferrer"
          >
            <span className="fa-stack fa-lg">
              <i className="fa fa-circle fa-stack-2x" />
              <i className="fa fa-linkedin fa-stack-1x fa-inverse" />
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
);

export default NavBar;
