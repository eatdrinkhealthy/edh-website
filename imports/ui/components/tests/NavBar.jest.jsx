/* eslint-env jest */
/* eslint-disable func-names, prefer-arrow-callback, no-unused-expressions */
import React from "react";
import renderer from "react-test-renderer";
import NavBar from "../NavBar";

describe("<NavBar />", function () {
  it("matches render snapshot", function () {
    const tree = renderer.create(<NavBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
