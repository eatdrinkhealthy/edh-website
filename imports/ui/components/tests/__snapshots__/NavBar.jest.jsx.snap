// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> matches render snapshot 1`] = `
<div
  className="row"
>
  <div
    className="column"
  >
    <div
      className="app-header__top"
    >
      <a
        href="/"
      >
        <img
          alt="Company Logo"
          src="images/logo.png"
        />
      </a>
      <nav
        className="app-header-nav"
      >
        <ul
          className="menu align-center"
        >
          <li>
            <a
              aria-current="true"
              className="active"
              href="/"
              onClick={[Function]}
              style={
                Object {
                  "color": "#f28020",
                }
              }
            >
              HOME
            </a>
          </li>
          <li>
            <a
              aria-current={false}
              className={undefined}
              href="#contactForm"
              onClick={[Function]}
              style={undefined}
            >
              CONTACT
            </a>
          </li>
          <li>
            <a
              aria-current={false}
              className={undefined}
              href="/privacy-policy"
              onClick={[Function]}
              style={undefined}
            >
              PRIVACY
            </a>
          </li>
        </ul>
      </nav>
      <div
        className="app-header-social text-center medium-text-right"
      >
        <a
          href="mailto:info@eatdrinkhealthy.co"
        >
          <span
            className="fa-stack fa-lg"
          >
            <i
              className="fa fa-circle fa-stack-2x"
            />
            <i
              className="fa fa-envelope fa-stack-1x fa-inverse"
            />
          </span>
        </a>
        <a
          href="https://www.linkedin.com/company/16173480/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            className="fa-stack fa-lg"
          >
            <i
              className="fa fa-circle fa-stack-2x"
            />
            <i
              className="fa fa-linkedin fa-stack-1x fa-inverse"
            />
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
